app/services/sso/jwt.rb

```diff
@@ -42,5 +48,10 @@ def bearer_token_set?
     def no_issuer?
       jwt_token.decoded_payload.present? && !jwt_token.decoded_payload.key?('iss')
     end
+
+    def valid_scope?(scope)
+      return true if scope.empty?
+      scope.split(' ').include? controller.controller_permission
```

Comment:
I still believe this should be limited to specific controller actions. When it comes to permissions, it's better to be explicit than implicit. 
For example, a token with `registration` scope could potentially allow an attacker to generate a new token with unlimited lifetime, since `/hosts/register` also has `controller_permission == 'registration'`. While it is indeed better than granting full access via the token, it is still a potential rights escalation.